packages/launchpad/cypress/e2e/integration/global-mode.spec.ts

```diff
@@ -0,0 +1,204 @@
+import defaultMessages from '@packages/frontend-shared/src/locales/en-US.json'
+import path from 'path'
+
+describe('Launchpad: Global Mode', () => {
+  describe('when no projects have been added', () => {
+    it('shows "Add Project" view', () => {
+      cy.openGlobalMode()
+      cy.visitLaunchpad()
+      cy.get('h1').should('contain', defaultMessages.globalPage.empty.title)
+      cy.get('[data-cy="dropzone"]')
+      .should('contain', defaultMessages.globalPage.empty.dropText.split('{0}')[0])
+      .find('button')
+      .should('contain', 'browse manually')
+    })
+
+    it('can add a project by dragging folder into project dropzone', () => {
+      cy.openGlobalMode()
+      cy.visitLaunchpad()
+      cy.get('h1').should('contain', defaultMessages.globalPage.empty.title)
+      cy.get('[data-cy="dropzone"]')
+      .should('contain', defaultMessages.globalPage.empty.dropText.split('{0}')[0])
+
+      cy.scaffoldProject('todos')
+      .then((projectPath) => {
+        cy.get('[data-cy="dropzone"]')
+        .dropFileWithPath(projectPath)
+      })
+
+      cy.contains('Welcome to Cypress!')
+      cy.get('a').contains('Projects').click()
+      cy.get('[data-cy="project-card"]')
+      .should('have.length', 1)
+      .should('contain', 'todos')
+    })
+  })
+
+  describe('when projects have been added', () => {
+    const setupProjects = (projectList) => {
+      cy.openGlobalMode()
+      projectList.forEach((projectName) => {
+        cy.addProject(projectName)
+      })
+
+      cy.visitLaunchpad()
+
+      cy.log('The recents list shows all projects that have been added')
+      cy.contains(defaultMessages.globalPage.recentProjectsHeader)
+
+      cy.get('[data-cy="project-card"]')
+      .should('have.length', projectList.length)
+      .each((card, index) => {
+        expect(card).to.contain(projectList[index])
+      })
+    }
+
+    it('shows the recent projects list sorted by most-recently opened', () => {
+      const projectList = ['todos', 'ids', 'cookies', 'plugin-empty']
+
+      setupProjects(projectList)
+    })
+
+    it('takes user to the next step when clicking on a project card', () => {
+      const projectList = ['todos']
+
+      setupProjects(projectList)
+      cy.get('[data-cy="project-card"]').click()
+      cy.get('[data-cy="project-card"]')
+      .should('have.length', 0)
+    })
+
+    it('updates most-recently opened project list when returning from next step', () => {
+      const projectList = ['todos', 'ids', 'cookies', 'plugin-empty']
+
+      setupProjects(projectList)
+
+      cy.get('[data-cy="project-card"]').within((cards) => cards.get(2).click())
+      cy.get('h1').contains('Welcome to Cypress!')
```

Comment:
The [testing-library cypress extensions](https://testing-library.com/docs/cypress-testing-library/intro/) are included by the support file. They'd allow you to write this assertion like:
```js
cy.findByRole('heading', { level: 1, name: 'Welcome to Cypress' }).should('be.visible') // visibility check optional we've leaned toward including these
```
This is slightly more verbose but a better abstraction from the DOM (while still verifying intent).

Reply:
You make a good point....but how much of our tests do we want to write with a 3rd-party library over our own commands?